libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -41,8 +38,8 @@ uint32_t ByteBuffer::available(void) const
 
 uint32_t ByteBuffer::space(void) const
 {
-    uint32_t _head;
-    return (((_head=head) > tail)?(_head - tail) - 1:((size - tail) + _head) - 1);
+    uint32_t _head = head;
+    return size ? (_head > tail ? 0 : size) + _head - tail - 1 : 0;
```

Comment:
I'd prefer this use of `?` operator to become an if else, since now it's much harder to follow.